### PR TITLE
Add glslang

### PR DESCRIPTION
**Pull request by satmandu**
Add glslang package which mesa looks for.

Works properly:
- [x] x86_64

### CODE REVIEW

File: packages/glslang.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Adapted from Arch Linux glslang PKGBUILD at:
+# https://github.com/archlinux/svntogit-packages/raw/packages/glslang/trunk/PKGBUILD
+
+require 'package'
+
+class Glslang < Package
+  description 'OpenGL and OpenGL ES shader front end and validator'
+  version '8.13.3743'
```
Comment (uberhacker) [inline packages/glslang.rb:8]: This is missing `compatibility 'all'`.

File: packages/glslang.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  source_url 'https://github.com/KhronosGroup/glslang/archive/8.13.3743.tar.gz'
+  source_sha256 '639ebec56f1a7402f2fa094469a5ddea1eceecfaf2e9efe361376a0f73a7ee2f'
+
+  depends_on 'python3'
+  depends_on 'cmake' => ':build'
+  depends_on 'ninja' => ':build'
```
Comment (uberhacker) [inline packages/glslang.rb:14]: These are all core packages so no need to list here.

File: packages/glslang.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    Dir.mkdir 'build-static'
+    Dir.chdir 'build-static' do
+    system "cmake .. \
+      -GNinja \
+      -DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX} \
+      -DCMAKE_INSTALL_LIBDIR=#{CREW_LIB_PREFIX} \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DBUILD_SHARED_LIBS=OFF"
+      end
+    system 'ninja -C build-static'
```
Comment (uberhacker) [inline packages/glslang.rb:39]: Any reason to build static libraries?

Comment (satmandu) [reply packages/glslang.rb:39]: It's cargo cult I know but I'm just copying logic from the arch pkgbuild file at https://github.com/archlinux/svntogit-packages/raw/packages/glslang/trunk/PKGBUILD

File: packages/glslang.rb (hunk 1 above)
Comment (satmandu) [reply packages/glslang.rb:8]: fixed

File: packages/glslang.rb (hunk 3 above)
Comment (uberhacker) [reply packages/glslang.rb:36]: I guess the minute you don't include it is the time you need it. LoL

**Pull request merged**